### PR TITLE
Post-Processing: Introducing base class for Passes

### PR DESCRIPTION
**Pull request by Mugen87**
This PR introduces `THREE.Pass`, a base class for post-processing passes. So far, i've added four properties and one method to this new entity. 
##### Properties

`enabled`: if set to true, the pass is processed by the composer
`needsSwap`: if set to true, the pass indicates to swap read and write buffer after rendering
`clear`: if set to true, the pass clears its buffer before rendering
`renderToScreen`: if set to true, the result of the pass is rendered to screen
##### Methods

`render( renderer, writeBuffer, readBuffer, delta, maskActive  )`
##### About the changes

All passes in three.js are now derived from this class. The changes of the PR are cautious. It was not necessary not rewrite code in the examples. `THREE.Pass` could be the basis for a more advanced interface (see #8148, ping @bhouston). 

As you can see, i've placed `THREE.Pass` not in the examples directory but in `src/extras/postprocessing/`. This is just a suggestion, but i think it would be good to include the new file in the core of the library. Otherwise, the dependency management of post-processing entities gets even more confusing. 

Maybe this could be the first step to generally include post-processing to the core? Just a thought...

### CODE REVIEW

File: utils/build/includes/extras.json [hunk 1]
```diff
@@ -65,5 +65,6 @@
 	"src/extras/helpers/VertexNormalsHelper.js",
 	"src/extras/helpers/WireframeHelper.js",
 	"src/extras/objects/ImmediateRenderObject.js",
-	"src/extras/objects/MorphBlendMesh.js"
+	"src/extras/objects/MorphBlendMesh.js",
+	"src/extras/postprocessing/Pass.js"
```
Comment (mrdoob) [inline utils/build/includes/extras.json]: Not sure about this. I would, instead, add the `THREE.Pass` code at the end of `EffectComposer.js`

**Comment by mrdoob** [comment]
> Maybe this could be the first step to generally include post-processing to the core? Just a thought...

I think I would prefer to keep it separate. Also, move it out of the `THREE` namespace, maybe it's own? `FX.Composer`, `FX.Pass`, ... ? Or maybe `THREE.FX.Composer`, `THREE.FX.Pass`, ... ?

**Comment by mrdoob** [comment]
Apart from the extras.json thing, this PR looks good though!

**Comment by Mugen87** [comment]
>  I would, instead, add the THREE.Pass code at the end of EffectComposer.js

I haven't thought of this option. The main point is that users don't need not include an additional JavaScript file to their project. So the last commit moved the code of `THREE.Pass` out of core to `EffectComposer.js`.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**